nearcore/src/config.rs

```diff
@@ -411,15 +420,18 @@ impl Config {
             );
         }
 
-        assert!(
-            config.archive || config.save_trie_changes,
-            "Configuration with archive = false and save_trie_changes = false is not supported \
-            because non-archival nodes must save trie changes in order to do do garbage collection."
-        );
-
+        config.validate()?;
```

Comment:
we have a small corner case here:

as we're doing the validation here (during the reading) - but later we pick only selected (upgradeable) fields to apply to our current config.

so we should re-run validate before applying the config changes to the new config.

in theory it can be done in a separate PR (as current validation doesn't look at upgrade'able fields)

Reply:
Sounds good, will follow-up. Added a TODO.